app/models/pydantic/creation_options.py

```diff
@@ -191,18 +191,50 @@ class VectorSourceCreationOptions(StrictBaseModel):
     )
     create_dynamic_vector_tile_cache: bool = Field(
         True,
-        description="By default, vector sources will implicitly create a dynamic vector tile cache. "
-        "Disable this option by setting value to `false`",
+        description=(
+            "By default, vector sources will implicitly create a dynamic vector tile cache. "
+            "Disable this option by setting value to `false`"
+        ),
     )
     add_to_geostore: bool = Field(
         True,
-        description="Include features to geostore, to make geometries searchable via geostore endpoint.",
+        description="Make geometries searchable via geostore endpoint.",
     )
     timeout: int = DEFAULT_JOB_DURATION
 
     @validator("source_uri")
     def validate_source_uri(cls, v, values, **kwargs):
-        if values.get("source_driver") != VectorDrivers.csv:
+        if values.get("source_driver") == VectorDrivers.csv:
+            assert len(v) >= 1, "CSV sources require at least one input file"
+        else:
+            assert (
+                len(v) == 1
+            ), "Non-CSV vector sources require one and only one input file"
+        return v
+
+
+class VectorSourceAppendOptions(StrictBaseModel):
+    source_type: VectorSourceType = Field(..., description="Source type of input file.")
```

Comment:
In the tabular append, I think we just accept source URIs and force it to use the same input options (i.e. be in the same format) as the data on initial creation. 

Does it make sense to allow potentially different formats go into the same table? I guess as long as the schema is the same it should work.

Reply:
And on the same note, should we actually be enforcing somewhere that the schema is the same as the original table, and return a 4XX error if not? Presumably it'll just fail later down the pipeline and you'd get a vague message in the task log. 

I doubt we check that now for tabular appends, and it'd be tricky since we only have a batch script for checking the schema (and may need to download the whole file for certain formats). So maybe this is more of a nice-to-have for later.

Reply:
> In the tabular append, I think we just accept source URIs and force it to use the same input options (i.e. be in the same format) as the data on initial creation.
> 
> Does it make sense to allow potentially different formats go into the same table? I guess as long as the schema is the same it should work.

Personally I'd like to allow things unless there's a specific reason not to. On the "for" side, I can imagine a situation in which updates start to be delivered in a different format than the original data at some point. But mostly I just dislike the idea of disallowing it without a strong reason to.

Reply:
> And on the same note, should we actually be enforcing somewhere that the schema is the same as the original table, and return a 4XX error if not? Presumably it'll just fail later down the pipeline and you'd get a vague message in the task log.
> 
> I doubt we check that now for tabular appends, and it'd be tricky since we only have a batch script for checking the schema (and may need to download the whole file for certain formats). So maybe this is more of a nice-to-have for later.

Mmm, good point. Since it's possible to specify the schema, we could require that if they specify in the original they need to specify in the append, and that they should match.
But you're right, it sounds like checking the actual schema of the data would at least sometimes (often?) require downloading the entire data file, which at least sometimes (often?) would cause a timeout.
Let me check, though. Maybe ogrinfo has some format-specific magic that can determine the schema of these files remotely without downloading the entire files. I'll investigate.

Reply:
On the other hand, requiring them to re-specify the schema when we don't really need/use it seems like it's adding an annoying hurdle. What do you think, @jterry64 ?

Reply:
So, plot twist: The way I wrote it didn't work. The append endpoint expects a JSON payload of (only)
{
  "source_uri": [some_uri(s)]
}
In order to support specifying a schema and such I would have to either make a breaking change to the API to or a special case for vector appends, which would be nasty. I don't like it, but I've made the vector append case consistent with the table append case. It's something I'd like to change when we increment API versions, though.